Add notifications when someone direct messages you

ahmedc3ca opened the pull request.

ahmedc3ca committed:
first working prototype

ahmedc3ca committed:
first working prototype

ahmedc3ca committed:
removed api key

ahmedc3ca committed:
removed api key

ahmedc3ca committed:
readded fcm key

ahmedc3ca committed:
readded fcm key

ahmedc3ca committed:
token uploaded but same token

ahmedc3ca committed:
token uploaded but same token

ahmedc3ca committed:
first testing attempt

ahmedc3ca committed:
first testing attempt

ahmedc3ca committed:
old tests almost passing

ahmedc3ca committed:
corrected database

ahmedc3ca committed:
Merge branch 'main' into feature/notifications

ahmedc3ca committed:
merging fixes

ahmedc3ca committed:
merging fixes

ahmedc3ca committed:
some tests

ahmedc3ca committed:
some tests

ahmedc3ca committed:
more tests

MarcelTorne commented on app/src/androidTest/java/com/github/epfl/meili/profile/friends/FriendsListActivityTest.kt:
```diff
@@ -113,13 +113,13 @@ class FriendsListActivityTest {
+//        Espresso.onView(textViewContainsText(TEST_FRIEND_NAME)).check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
+//
+//        Espresso.onView(buttonViewContainsText("Send")).check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
     }
```
Why is this test being commented out?

ahmedc3ca replied:
Hi, in this test the chat log activity is opened. Since the added database isn't mocked in the test, it was failing. I will add the mock now and uncomment this part.

MarcelTorne commented on app/src/main/java/com/github/epfl/meili/notifications/FirebaseNotificationService.kt:
```diff
@@ -0,0 +1,105 @@
+        //clearing all intents (start fresh)
+        //intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+        //intent can only be used once
+        //val pendingIntent = PendingIntent.getActivity(this, 0, intent, FLAG_ONE_SHOT)
```
Same here?

MarcelTorne replied:
Sounds good thanks!

MarcelTorne reviewed (changes requested):
Hi Ahmed! Thanks a lot for your work. I have added a couple of comments regarding commented code. I will review it again afterwards. Thanks again I think this will be a really cool feature!

ahmedc3ca committed:
Merge branch 'feature/notifications' of github.com:meili-epfl/Meili into feature/notifications

# Conflicts:
#	app/build.gradle
#	app/src/androidTest/java/com/github/epfl/meili/notifications/FirebaseNotificationServiceTest.kt
#	app/src/main/AndroidManifest.xml
#	app/src/main/java/com/github/epfl/meili/database/Database.kt
#	app/src/main/java/com/github/epfl/meili/messages/ChatLogActivity.kt
#	app/src/main/java/com/github/epfl/meili/util/MeiliViewModel.kt

ahmedc3ca committed:
Merge remote-tracking branch 'origin/feature/notifications' into feature/notifications

# Conflicts:
#	app/build.gradle
#	app/src/main/AndroidManifest.xml
#	app/src/main/java/com/github/epfl/meili/database/Database.kt
#	app/src/main/java/com/github/epfl/meili/database/FirestoreDatabase.kt
#	app/src/main/java/com/github/epfl/meili/messages/ChatLogActivity.kt
#	app/src/main/java/com/github/epfl/meili/util/MeiliViewModel.kt

ahmedc3ca committed:
uncommented tests + added comments on removed intent code

ahmedc3ca committed:
clearer comment

ahmedc3ca committed:
Merge branch 'main' into feature/notifications

ahmedc3ca committed:
Merge branch 'main' into feature/notifications

tberkane committed:
Merge branch 'main' into feature/notifications

ahmedc3ca committed:
more tests

ahmedc3ca committed:
Merge branch 'main' into feature/notifications

ahmedc3ca committed:
fix merge with main

ahmedc3ca committed:
reduce MapActivity.kt size

ahmedc3ca committed:
change checking

MarcelTorne committed:
Merge branch 'main' into feature/notifications

MarcelTorne reviewed (approved):
Looks good to me! Unfortunately I cannot test it but I trust that you did test it and that its working. Thanks for bringing this!

aman3014 commented on app/src/main/java/com/github/epfl/meili/notifications/FirebaseNotificationService.kt:
```diff
@@ -0,0 +1,124 @@
+        super.onMessageReceived(message)
+
+        Log.d("notif", "received message notification")
+        //Uncomment this code to make clicking notifications open activity
```
Could you add an explanation of why we couldn't/chose to not do it.

ahmedc3ca replied:
I added the things needed to make the commented code work. Maybe this way it's clearer why I couldn't do it 👍 .

aman3014 commented on app/src/main/java/com/github/epfl/meili/messages/ChatActivity.kt:
```diff
@@ -125,15 +139,59 @@ class ChatActivity : PoiActivity(R.layout.activity_chat, R.id.chat_activity) {
+
     private fun performSendMessage() {
         val text = findViewById<EditText>(R.id.edit_text_chat_log).text.toString()
+        if (text.isEmpty()) {
+            return
+        }
```
Nice addition!

aman3014 commented on .gitignore:
```diff
@@ -9,3 +9,4 @@
 .cxx
 local.properties
 app/src/main/res/values/keys.xml
+app/src/main/res/values/keys.xml
```
Duplicate

aman3014 commented on app/.gitignore:
```diff
@@ -1,2 +1,3 @@
 /build
 google-services.json
+google-services.json
```
Duplicate

aman3014 reviewed (commented):
The code looks good! I left a few comments.

Also, could you remove the `Log.d`s in chat activity if they are not necessary?

ahmedc3ca committed:
pr comments

ahmedc3ca commented:
> The code looks good! I left a few comments.
> 
> Also, could you remove the `Log.d`s in chat activity if they are not necessary?

Thank you for the feedback! I addressed the comments in the last commit. Did you get to test the feature again on your device?

aman3014 commented:
> > The code looks good! I left a few comments.
> > Also, could you remove the `Log.d`s in chat activity if they are not necessary?
> 
> Thank you for the feedback! I addressed the comments in the last commit. Did you get to test the feature again on your device?

I couldn't test actually, were u able to test on the emulator?

ahmedc3ca committed:
added To be implemented in the future comment

aman3014 reviewed (approved):
We discussed testing offline.

Thanks for this feature!

aman3014 committed:
Merge branch 'main' into feature/notifications
